src/providers/proxy/proxy_id.c

```diff
@@ -1488,6 +1492,153 @@ static int get_initgr(TALLOC_CTX *mem_ctx,
+
+    tmp_ctx = talloc_new(NULL);
+    if (!tmp_ctx) {
+        ret = ENOMEM;
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc_new() failed\n");
+        goto done;
+    }
+
+    username = sss_create_internal_fqname(tmp_ctx, pwd->pw_name, dom->name);
+    if (username == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "Failed to create fqdn '%s'\n", pwd->pw_name);
+        ret = ENOMEM;
+        goto done;
+    }
+
+    grp = talloc(tmp_ctx, struct group);
+    if (!grp) {
+        ret = ENOMEM;
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc() failed\n");
+        goto done;
+    }
+
+    for (i = 0; i < num_cached_gids; i++) {
+        group_found = false;
+        // group 0 is the primary group so it can be skipped
+        for (j = 1; j < num_gids; j++) {
+            if (cached_gids[i] == gids[j]) {
+                group_found = true;
+                break;
+            }
+        }
+
+        if (!group_found) {
+            do {
+                /* always zero out the grp structure */
+                memset(grp, 0, sizeof(struct group));
+                buffer = grow_group_buffer(tmp_ctx, &buffer, &buflen);
+                if (!buffer) {
+                    ret = ENOMEM;
+                    break;
+                }
+
+                status = ctx->ops.getgrgid_r(cached_gids[i], grp, buffer, buflen, &ret);
+
+                ret = handle_getgr_result(status, grp, dom, NULL);
+            } while (ret == EAGAIN);
+
+            if (ret != EOK) {
+                DEBUG(SSSDBG_OP_FAILURE,
+                      "getgrgid failed [%d]: %s\n", ret, strerror(ret));
+                continue;
+            }
+
+            groupname = sss_create_internal_fqname(tmp_ctx, grp->gr_name, dom->name);
+            if (groupname == NULL) {
+                DEBUG(SSSDBG_OP_FAILURE, "Failed to create fqdn '%s'\n",
+                      grp->gr_name);
+                ret = ENOMEM;
+                continue;
+            }
+
+            ret = sysdb_remove_group_member(dom, groupname,
+                                            username,
+                                            SYSDB_MEMBER_USER, false);
+
+            if (ret != EOK) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "Could not remove member [%s] from group [%s]\n",
+                      username, groupname);
+                continue;
+            }
+        }
+    }
+
+done:
+    return ret;
+}
+
+static int get_cached_user_members(struct sysdb_ctx *sysdb,
+                                   struct sss_domain_info *dom,
+                                   const struct passwd *pwd,
+                                   unsigned int *_num_cached_gids,
+                                   gid_t **_cached_gids)
+{
+    TALLOC_CTX *tmp_ctx = NULL;
+    int ret = EOK;
+    int i = 0, j = 0;
+    gid_t gid = 0;
+    gid_t *cached_gids = NULL;
+    const char *username = NULL;
+    struct ldb_result *res = NULL;
+    unsigned int num_cached_gids = 0;
+
```

Comment:
I don't think that's going to work. `_cached_gids` should be NULL when entering this function. As for `_num_cached_gids`, currently this is a stacked variable so I don't think it's necessary to check that it isn't NULL. If you are proposing the change to be on the safe side, just in case somebody else calls this function then I'm fine with it.